server/src/main/java/org/opensearch/indices/NodeIndicesStats.java

```diff
@@ -112,6 +113,63 @@ public NodeIndicesStats(CommonStats oldStats, Map<Index, List<IndexShardStats>>
         }
     }
 
+    public NodeIndicesStats(
+        CommonStats oldStats,
+        Map<Index, List<IndexShardStats>> statsByShard,
+        SearchRequestStats searchRequestStats,
+        String[] levels
+    ) {
+        // make a total common stats from old ones and current ones
+        this.stats = oldStats;
+        for (List<IndexShardStats> shardStatsList : statsByShard.values()) {
+            for (IndexShardStats indexShardStats : shardStatsList) {
+                for (ShardStats shardStats : indexShardStats.getShards()) {
+                    stats.add(shardStats.getStats());
+                }
+            }
+        }
```

Comment:
If we are computing `statsByIndex` - can we compute in same loop ?

Reply:
Makes sense, will update.

Reply:
Looking back, I noticed that while computing statsByIndex, we are iterating over each key in the map and accordingly making the Index Level CommonStats entry. However for node level CommonStats, we are simply going over each value in the hash-map irrespective of the key. 

I think keeping the computation on the same loop will just add more complexity as well as computational effort when level is not provided. Kept it separate itself for now